main.nf

```diff
@@ -127,95 +179,81 @@ process run_cell_types_eval {
 	memory { 16.GB * task.attempt }
 	
 	input:
-	    tuple val(fold), file(test), file(train) from SPLIT_DATA
-	    file(marker_genes) from MARKERS
-	    val(dataset_id) from DATASET_ID 
+	tuple val(fold), file(test_TPM), file(test_CPM), file(train_TPM), file(train_CPM), val(dataset_id) from COMB_DATA
```

Comment:
This logic seems to indicate that you will always need TPMs and CPMs. That might be a reasonable simplifying assumption to make, but would mean you don't need the logic above which selects metrics based on activated method.

Reply:
What would happen if I only activated methods which required CPM, for example? This process would lack required inputs wouldn't it?

Reply:
In case only one method is enabled, say CPM, `test_TPM` and `train_TPM` would still contain CPM data. In this case it's a bit tricky to read as variables are labeled, but there is no error when running only one type of matrix.

Reply:
Sorry, actually there is an error, because: **Input tuple does not match input set cardinality**